scripts/zones/Sealions_Den/helpers/One_to_be_Feared.lua

```diff
@@ -0,0 +1,141 @@
+    local mammetOffset = ID.mob.ONE_TO_BE_FEARED_OFFSET + (7 * (battlefield:getArea() - 1))
+    local mammetDeathCount = 0
+
+    -- If all five mammets in this instance are dead, start event.
+    for i = mammetOffset + 0, mammetOffset + 4 do
+        if GetMobByID(i):isDead() then
+            mammetDeathCount = mammetDeathCount + 1
+            if mammetDeathCount == 5 then
+                player:startEvent(10)
+            end
+        else
+            break
+        end
+    end
+end
+
+xi.one_to_be_feared.handleMammetBattleEnding = function(player, csid, option)
+    if csid == 10 then
+        -- Players are healed in between fights, but their TP is set to 0
+        player:addTitle(xi.title.BRANDED_BY_LIGHTNING)
+        healCharacter(player)
+
+        -- Move player to instance start. End battle 1.
+        player:setLocalVar("[OTBF]battleCompleted", 1)
+        returnToAirship(player)
+    end
+end
+
+-----------------------------------
+-- Battle 2: Omega
+-----------------------------------
+xi.one_to_be_feared.handleOmegaDeath = function(mob, player, isKiller)
+    player:startEvent(11)
+end
+
+xi.one_to_be_feared.handleOmegaBattleEnding = function(player, csid, option)
+    if csid == 11 then
+        -- Players are healed in between fights, but their TP is set to 0.
+        player:addTitle(xi.title.OMEGA_OSTRACIZER)
+        healCharacter(player)
+
+        -- Move player to instance start. End battle 2.
+        player:setLocalVar("[OTBF]battleCompleted", 2)
+        returnToAirship(player)
+    end
+end
+-----------------------------------
+-- Battle 3: Ultima
+-----------------------------------
+xi.one_to_be_feared.handleUltimaDeath = function(mob, player, isKiller)
+    player:addTitle(xi.title.ULTIMA_UNDERTAKER)
+    player:setLocalVar("[OTBF]battleCompleted", 0)
+end
+
+-----------------------------------
+-- While on Airship
+-----------------------------------
+xi.one_to_be_feared.handleAirshipDoorTrigger = function(player, npc)
+    player:startEvent(32003, npc:getID() - ID.npc.AIRSHIP_DOOR_OFFSET + 1, player:getLocalVar("[OTBF]battleCompleted") * 2)
+end
+
+xi.one_to_be_feared.handleOnEventUpdate = function(player, csid, option)
+    local battlefield = player:getBattlefield()
+
+    -- Spawn Omega for given instance.
+    if csid == 1 and option == 0 then
+        local omegaId = ID.mob.ONE_TO_BE_FEARED_OFFSET + (7 * (battlefield:getArea() - 1)) + 5
+        if omegaId and not GetMobByID(omegaId):isSpawned() then
+            SpawnMob(omegaId)
+        end
+
+    -- Spawn Ultima for given instance.
+    elseif csid == 2 and option == 0 then
+        local ultimaId = ID.mob.ONE_TO_BE_FEARED_OFFSET + (7 * (battlefield:getArea() - 1)) + 6
+        if not GetMobByID(ultimaId):isSpawned() then
+            SpawnMob(ultimaId)
+            battlefield:setLocalVar("phaseChange", 0)
+        end
+
+    -- Leave battlefield.
+    elseif csid == 32003 and option == 2 then
+        player:updateEvent(3)
+        return true
```

Comment:
Doesn't look like we're using this and the below return value anywhere, probably not needed